### ISSUE TITLE
how to use with emacs

### ISSUE THREAD

**Issue by KeithPiTsui** [Decodetalkers/neocmakelsp#64]
I am new to Emacs, and I wonder how to use neocmakelsp with emacs.

**Comment by Decodetalkers** [comment]
https://github.com/manateelazycat/lsp-bridge/blob/master/langserver/sumneko.json maybe you can use this repo, or make a pr to it, I do not know much about emacs, sorry

**Comment by yangyingchao** [comment]
Here is my configuration for using `neocmakelsp` with `eglot` in `cmake-ts-mode`:

``` emacs-lisp
(use-package cmake-ts-mode
  :config
  (add-hook 'cmake-ts-mode-hook
    (defun yc/cmake-mode-hook ()
      (push `((cmmake-ts-mode) . ("neocmakelsp" "--stdio")) eglot-server-programs)
      (eglot-ensure))))
```

Please change `cmake-ts-mode` to `cmake-mode` if you are using `cmake-mode`.

**Comment by Decodetalkers** [comment]
Has your problem been solved?

**Comment by SL-RU** [comment]
Hi, any progress?

**Comment by Decodetalkers** [comment]
> Hi, any progress?

You can read the readme, there is an example config

**Comment by joaotavora** [comment]
Eglot author here.  Just go the cmake file and `C-u M-x eglot`.  In the prompt, type `neocmakelsp --stdio`.  That's it.  If you need to put something in your config file, put just this:

```
(add-to-list 'eglot-server-programs `((cmake-mode cmake-ts-mode) . ("neocmakelsp" "--stdio")))
```

Next time, just type `M-x eglot` for each project.

A bit tired of this `eglot-ensure` and `use-package` cargo-culting.

**Comment by joaotavora** [comment]
Yes, just building the AST isn't enough, you have to understand language scoping rules, and with cmake it's crazy hard.  I don't know these rules myself (this is why I'm looking for a LSP to teach me about them).   You might get the Cmake guys to help you though.  I think Cmake usage would benefit immensely from having a proper LSP. Are you in the cpplang slack (https://cppalliance.org/slack/) by any chance?  It would be a good way to connect you guys perhaps.

**Comment by Decodetalkers** [comment]
> Yes, just building the AST isn't enough, you have to understand language scoping rules, and with cmake it's probably hard (or maybe not?). I don't know these rules myself (this is why I'm looking for a LSP to teach me about them). You might get the Cmake guys to help you though. I think Cmake usage would benefit immensely from having a proper LSP. Are you in the cpplang slack (https://cppalliance.org/slack/) by any chance? It would be a good way to connect you guys perhaps.

Thank you, I have already join that workspace. Now I am there. You can find me buy search my id.

**Comment by Decodetalkers** [comment]
> Yes, just building the AST isn't enough, you have to understand language scoping rules, and with cmake it's probably hard (or maybe not?). I don't know these rules myself (this is why I'm looking for a LSP to teach me about them). You might get the Cmake guys to help you though. I think Cmake usage would benefit immensely from having a proper LSP. Are you in the cpplang slack (https://cppalliance.org/slack/) by any chance? It would be a good way to connect you guys perhaps.

https://github.com/neocmakelsp/neocmakelsp/pull/146
rename is finished
